### PR TITLE
Fix mem::uninitialized depricated since 1.38

### PR DESCRIPTION
**Pull request by superriva**
Fix for https://github.com/CraneStation/cranelift/issues/826.

### CODE REVIEW

**Comment by superriva** [comment]
Errors from Travis-CI:
```
$ rustc --version
rustc 1.35.0 (3c235d560 2019-05-20)
...
   Compiling cranelift-simplejit v0.33.0 (/home/travis/build/CraneStation/cranelift/cranelift-simplejit)
error[E0658]: use of unstable library feature 'maybe_uninit' (see issue #53491)
  --> cranelift-simplejit/src/memory.rs:34:46
   |
34 |             let mut ptr: *mut libc::c_void = mem::MaybeUninit::uninit().assume_init();
   |                                              ^^^^^^^^^^^^^^^^^^^^^^^^
   Compiling cranelift-filetests v0.33.0 (/home/travis/build/CraneStation/cranelift/cranelift-filetests)
error[E0658]: use of unstable library feature 'maybe_uninit' (see issue #53491)
  --> cranelift-simplejit/src/memory.rs:34:73
   |
34 |             let mut ptr: *mut libc::c_void = mem::MaybeUninit::uninit().assume_init();
   |                                                                         ^^^^^^^^^^^
error: aborting due to 2 previous errors
...
```

File: cranelift-simplejit/src/memory.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ impl PtrLen {
         let page_size = region::page::size();
         let alloc_size = round_up_to_page_size(size, page_size);
         unsafe {
-            let mut ptr: *mut libc::c_void = mem::uninitialized();
+            let mut ptr: *mut libc::c_void = mem::MaybeUninit::uninit().assume_init();
```
Comment (bjorn3) [inline cranelift-simplejit/src/memory.rs]: This is wrong. Calling `assume_init` on an uninitialized `MaybeUninit` is UB.

```suggestion
            let mut ptr: *mut libc::c_void = 0 as *mut _;
```

**Comment by superriva** [comment]
There's recommendation to replace `0 as *const/mut T` with `ptr::null/null_mut()`:
https://github.com/rust-lang/rust/pull/21444

**Comment by bjorn3** [comment]
Forgot about that :)

**Comment by superriva** [comment]
@bnjbvr , 3 commits have been squashed.

**Comment by superriva** [comment]
@bjorn3 , @bnjbvr , Thank you.

**Pull request merged**